onnxruntime/core/providers/cpu/tensor/gather.cc

```diff
@@ -14,8 +14,8 @@ namespace op_kernel_type_control {
 ORT_SPECIFY_OP_KERNEL_ARG_DEFAULT_TYPES_ALL_OPSETS(
     kCpuExecutionProvider, kOnnxDomain, Gather, Input, 1, int32_t, int64_t);
 ORT_SPECIFY_OP_KERNEL_ARG_REQUIRED_TYPES_ALL_OPSETS(
-    kCpuExecutionProvider, kOnnxDomain, Gather, Input, 1, int64_t);
-}
+    kCpuExecutionProvider, kOnnxDomain, Gather, Input, 1, int32_t, int64_t);
```

Comment:
>int32_t, int64_t [](start = 58, length = 16)

if the default and required types are the same, there's not really a reason to have the type reduction here at all. but it's fine to clean that up later too

Reply:
same comment for Slice input 1

---
In reply to: [605087765](https://github.com/microsoft/onnxruntime/pull/7192#discussion_r605087765) [](ancestors = 605087765)